File: loop_lambda.js
function make_functions() {
  var functions = [];
  var i = 0;
  while(i < 10) {
    functions.push(function(x) { return x + i; });
    i += 1;
  }
  return functions;
}

var functions = make_functions();
var answers = [];
for(var j = 0; j < functions.length; j += 1) {
  answers.push(functions[j](100));
}

console.log(answers);

File: loop_lambda.py
def make_functions():
  functions = []
  i = 0
  while i < 10:
    functions.append(lambda x: x + i)
    i += 1
  return functions

functions = make_functions()
answers = []
j = 0
while j < 10:
  answers.append(functions[j](100))
  j += 1

print answers
